### ISSUE TITLE
Preventing highlighting for selected texts doesn't work properly

### ISSUE THREAD

**Issue by hiiwave** [jcb91/jupyter_highlight_selected_word#10]
As described in #9, this feature does not work to me. It seems that `CodeMirror-selectedtext` isn't applied properly to selected texts under my chrome browser.
('seems' means that I'm not sure whether it's true. I could not observe this property well under chrome's devtool since it changes whenever I switched my mouse from the selected texts to the devtool window.)
Do you have ideas how could I verify this or any workarounds?

**Comment by hiiwave** [comment]
Here's my screen when I select the second `hello`.
![0](https://cloud.githubusercontent.com/assets/5608511/23914127/fbdd87a6-091f-11e7-8d55-e85baa70c6ab.png)

I also wonder though I could actually see the three `hello` switches to new background colors, the "Styles" window (right hand side) in chrome's devtool doesn't display it. It makes me hard to figure out what happens here. I would really appreciate if you could share some experiences with that :)

**Comment by jcb91** [comment]
> ('seems' means that I'm not sure whether it's true. I could not observe this property well under chrome's devtool since it changes whenever I switched my mouse from the selected texts to the devtool window.)

Yes, this is a problem I encountered myself, and one for which I haven't yet thought of a solution (I just fiddled until I got things working as I wanted, and looked at CodeMirror css source for inspiration). The only simple diagnostic I can think of right now is to add some extra javascript to keep polling the page for existence of elements matching the `.CodeMirror-selectedtext` selector, and change some property (e.g. page background color) as a super-obvious indication of their presence/absence.

> It seems that CodeMirror-selectedtext isn't applied properly to selected texts under my chrome browser.

I notice that you seem to be using a non-vanilla theme, which _may_ be relevant here, although it also may not be. The class is applied to selected text by the codemirror addon [mark-selection](), which jupyter_highlight_selected_word loads using requirejs with its _default settings_, which means the class applied is the default `CodeMirror-selectedtext`. If the mark-selection addon has been loaded using a classname other than the default `CodeMirror-selectedtext`, this would cause the behaviour you describe. For diagnosis, could you try the following in the dev console?

```javascript
require([
    'codemirror/addon/selection/mark-selection'
], function on_success () {
    console.log('[mark-selection]', 'successful load.');
    console.log('[mark-selection]', 'styleSelectedText =',
                CodeMirror.getOption('styleSelectedText'));
}, function on_fail (reason) {
    console.error('[mark-selection]', reason);
});
```

I suppose it should be possible to query the option in jupyter_highlight_selected_word, then modify the css rule from javascript to suit the applied classname...

**Comment by hiiwave** [comment]
Thanks for taking time addressing this! I've checked and figured out that my observation was not right. In fact this feature works when I select a "range" (i.e. when `cm.somethingSelected() == true`), and does not work as I expect when I only put my cursor onto a word. 

The above three hello example I put was the situation that I "put" my cursor onto the second hello instead of actually selecting it. Hence `.CodeMirror-selectedtext` isn't applied there. I noticed that I could disable highlighting when nothing is selected by setting the `show_token = ''`, and I'm not bothered by the unexpected highlighting while I'm typing something new now.

Great thanks again for your concern! The only concern remained is when I select a word, it highlights not only matched whole word but also partially matched word.  Let me elaborate this in another issue.
